src/serialization/archive.hpp

```diff
@@ -18,6 +18,17 @@
 #include <boost/archive/binary_iarchive.hpp>
 #include <boost/archive/binary_oarchive.hpp>
 
+/* undo some macro defs in pyport.h */
```

Comment:
This fix should not appear here if it is only related to Python compilation issues.
It may be dangerous for projects which depend on Pinocchio to under some many macros like that.

Reply:
It is related to Python, but this is the only place where I found it was working. I could not just put it in the bindings. I can take a closer look and see if I can manage to change this.

Reply:
You should be able to put them in the right place. You should also be able to active this fix for certain Python versions, not all.

Reply:
I will check. By the way, do you have the problem, or not?

Reply:
Not at all. It is passes on all the Ubuntu versions of GitLab + Travis + MacOS Mojave

Reply:
Any news?

Reply:
Sorry, haven't had time to work on it. Are you still interested in this fix?

Reply:
Sure. If it allows using Pinocchio on a large spectrum of OS.

Reply:
I fixed this issue from my side, I will provide a fix soon.
It is a matter of headers inclusion.